### ISSUE TITLE
This Tool Has Incorrect Data Bug?

### ISSUE THREAD

**Issue by jimmytrouser** [SlimeKnights/IguanaTweaksTConstruct#64]
I am using the latest:
IguanaTinkerTweaks-1.7.10-2.1.1
TConstruct-1.7.10-1.8.1

I made a flint pickaxe (wood binding and handle)  and the first bonus that was put on it was a diamond, and I got the message to place the tool in a crafting window to update; I do this but nothing happens.

Below are the configs I used:

(main.cfg)

# Configuration file

##########################################################################################################
# allowedtools
#--------------------------------------------------------------------------------------------------------#
# Tweak Module: This category allows you to specify which tools ARE NOT USABLE or alternatively ARE STILL USABLE if the option to disable non-TConstsruct tools is enabled.
# To make this easier a /dumpTools command is provided, that dumps the names of all applicable items in your world. Copy'n'Paste away!
##########################################################################################################

allowedtools {
    # Bows that are excluded if the option to nerf non-tinkers bows is enabled. [default: [ThermalExpansion:tool.bowInvar]]
    S:bows <
        
     >

    # Change the type of the exclusion.
    # 'blacklist' means the listed tools are made unusable.
    # 'whitelist' means ALL tools except the listed ones are unusable. [default: blacklist]
    S:exclusionType=whitelist

    # Hoes that are excluded if the option to nerf non-tinkers hoes is enabled. [default: [Steamcraft:hoeGildedGold], [Steamcraft:hoeBrass], [ThermalExpansion:tool.hoeInvar], [IC2:itemToolBronzeHoe], [Railcraft:tool.steel.hoe]]
    S:hoes <
        
     >

    # Here you can exclude entire mods by adding their mod-id (the first part of the string). [default: [minecraft], [Metallurgy], [Natura], [BiomesOPlenty], [ProjRed|Exploration], [appliedenergistics2], [MekanismTool]]
    S:mods <
        Thaumcraft
        Botania
        ThaumicTinkerer
        TwilightForest
        EnderIO
        ExtraUtilities
        EnderZoo
        gendustry
        
     >

    # Swords that are excluded if the option to nerf non-tinkers swords is enabled. [default: [Botania:manasteelSword], [Steamcraft:swordGildedGold], [Steamcraft:swordBrass], [ThermalExpansion:tool.swordInvar], [IC2:itemToolBronzeSword], [Railcraft:tool.steel.sword]]
    S:swords <
        
     >

    # Tools that are excluded if the option to nerf non-tinkers tools is enabled. [default: [Botania:manasteelAxe], [Botania:manasteelPick], [Botania:manasteelShovel], [Steamcraft:axeGildedGold], [Steamcraft:pickGildedGold], [Steamcraft:shovelGildedGold], [Steamcraft:axeBrass], [Steamcraft:pickBrass], [Steamcraft:shovelBrass], [ThermalExpansion:tool.axeInvar], [ThermalExpansion:tool.pickaxeInvar], [ThermalExpansion:tool.shovelInvar], [IC2:itemToolBronzeAxe], [IC2:itemToolBronzePickaxe], [IC2:itemToolBronzeSpade], [Railcraft:tool.steel.axe], [Railcraft:tool.steel.pickaxe], [Railcraft:tool.steel.shovel]]
    S:tools <
        Railcraft:tool.crowbar
        Railcraft:tool.crowbar.magic
        Railcraft:tool.crowbar.reinforced
        
     >
}

##########################################################################################################
# debug
#--------------------------------------------------------------------------------------------------------#
# Stuff to give you/me more information
##########################################################################################################

debug {
    # Logs when the harvest level of a block is changed [default: true]
    B:logBlockHarvestLevelChange=true

    # Logs how much the extra-chance from doing stuff you had when getting a random bonus on levelup. [default: true]
    B:logBonusExtraChance=true

    # Logs every single thing done by the Override module. Use at your own risk. ;) [default: false]
    B:logExcessiveOverrideChanges=false

    # Logs when the mining level of a tinkers tool material is changed [default: true]
    B:logTinkerMaterialChange=true

    # Logs when the mining level of a (non-tinker) tool is changed [default: true]
    B:logToolMiningLevelChange=true

    # Current Tool/Pick XP is shown as debug (F3) text [default: false]
    B:showDebugXP=false
}

##########################################################################################################
# harvestleveltweaks
#--------------------------------------------------------------------------------------------------------#
# Harvest Level Tweak Module: Introduces a slower mining level progression.
##########################################################################################################

harvestleveltweaks {
    # Changes the Diamond and Emerald modifier: Apply it to a bronze level tool to obtain diamond level. Required unless you have steel or similar. [default: true]
    B:diamondRequired=true

    # Change durability of all tool materials (in percent) [range: 1 ~ 999, default: 80]
    I:durabilityPercentage=80

    # Change mining speed of all tool materials (in percent) [range: 1 ~ 999, default: 100]
    I:miningSpeedPercentage=100
}

##########################################################################################################
# mobheads
#--------------------------------------------------------------------------------------------------------#
# Mob Head Module: Adds additional Mob heads and drops
##########################################################################################################

mobheads {
    # Base percentage for a head to drop [range: 1 ~ 100, default: 5]
    I:baseDropChange=5

    # Percentage added to base percentage per level of Beheading modifier [range: 1 ~ 100, default: 2]
    I:beheadingDropChange=2
}

partreplacement {
    # How much of the current XP% to the next mining level shall be removed when replacing the pickaxe head. Useful to remove the mining level boost on part replacement. [range: 0 ~ 100, default: 5]
    I:PickBoostXpPenality=5

    # How much of the current XP% shall be removed when replacing parts (So if you had 50%, and penality is 10% it'll remove 5% xp, resulting in 45%). Does not remove Skill Levels. [range: 0 ~ 100, default: 0]
    I:XpPenality=0

    # Removes the Mob Head Modifier on Tool-Head replacement, allowing it to be reapplied. Should be used with PickBoostXpPenality. [default: true]
    B:removeMobHead=true
}

##########################################################################################################
# pickleveling
#--------------------------------------------------------------------------------------------------------#
# Leveling Module: Allows pickaxes to gain a mining level with enough XP. Should be used with the HarvestLevel Module.
##########################################################################################################

pickleveling {
    # Mob heads can be used to boost a pickaxe's mining xp. [default: true]
    B:addMobHeadBoost=true

    # Pickaxes gain Mining Xp by using the pickaxe. [default: true]
    B:allowLevelingBoost=true

    # Mob head boosting requires a free modifier [default: false]
    B:mobHeadBoostNeedsModifier=false

    # Every Pickaxes Mining Level is reduced by 1 and needs a mining levelup (separate from tool level) or, if enabled, a mob head modifier to advance [default: true]
    B:pickaxeBoostRequired=true

    # Change the percentage of XP required to boost a pick (i.e. 200 means 2x normal boost xp required) [range: 1 ~ 999, default: 100]
    I:xpRequiredPickBoostPercentage=100
}

##########################################################################################################
# randombonuses
#--------------------------------------------------------------------------------------------------------#
# Leveling Module: Allows to completely deactivate specific modifiers on levelup.
##########################################################################################################

randombonuses {
    B:allowAttack=true
    B:allowAutosmelt=true
    B:allowBaneOfArthropods=true
    B:allowBeheading=true
    B:allowCritical=true
    B:allowDiamond=true
    B:allowEmerald=true
    B:allowFiery=true
    B:allowJagged=true
    B:allowKnockback=true
    B:allowLifeSteal=true
    B:allowLuckLooting=true
    B:allowRedstone=true
    B:allowReinforced=true
    B:allowRepair=true
    B:allowSilkTouch=true
    B:allowSmite=true
    B:allowStonebound=true
}

##########################################################################################################
# toolleveling
#--------------------------------------------------------------------------------------------------------#
# Leveling Module: Setup the leveling system how you like it
##########################################################################################################

toolleveling {
    # Adds a random bonus on these levelups if 'RandomBonuses' is enabled
    I:BonusesAtLevels <
        2
        3
        4
        5
        6
     >

    # Each modifier is equally likely on levelup. Disables useful bonuses. [default: false]
    B:CompletelyRandomBonuses=false

    # The amount of modifiers new tools have. [range: 0 ~ 9, default: 0]
    I:ExtraModifiers=0

    # Adds an extra modifier on these levelups if 'ExtraModifiers' is enabled
    I:ModifiersAtLevels <
        2
        4
        6
     >

    # Gives a random bonus every level, if false and levelling is on modifiers are given at levels 2 and 4 (requires 'toolLeveling=true') [default: true]
    B:RandomBonuses=true

    # Disables less-useful modifiers on levelups. Like a sword with silktouch, or a pickaxe with beheading. [default: true]
    B:UsefulBonuses=true

    # XP tooltip shows numbers, in addition to percentage [default: true]
    B:detailedXpTooltip=true

    #  [range: 1 ~ 99, default: 6]
    I:maxToolLevel=6

    # Current XP% is shown after the level [default: false]
    B:showMinimalTooltipXP=false

    # Current XP is shown when hovering over a tool [default: true]
    B:showTooltipXP=true

    # Can your skill with tools 'level up' as you use them? [default: true]
    B:toolLeveling=true

    # Exponential multiplier for required boost xp per level [range: 1.0 ~ 9.99, default: 1.12]
    S:xpPerBoostLevelMultiplier=1.12

    # Exponential multiplier for required xp per level [range: 1.0 ~ 9.99, default: 1.15]
    S:xpPerLevelMultiplier=1.15

    # Change the XP required to level up tools in % (higher = more xp needed) [range: 1 ~ 999, default: 100]
    I:xpRequiredToolsPercentage=100

    # Change the XP required to level up weapons in % (higher = more xp needed) [range: 1 ~ 999, default: 100]
    I:xpRequiredWeaponsPercentage=100
}

##########################################################################################################
# tweaks
#--------------------------------------------------------------------------------------------------------#
# Tweak Module: Tweaks to vanilla Minecraft and Tinker's Construct
##########################################################################################################

tweaks {
    # Adds a shapeless recipe to get flint from gravel [default: true]
    B:addFlintRecipe=true

    # Allows toolparts to be used as material in the Part Builder. Like, turn a Pick head into a Shovel head.! [default: true]
    B:allowPartReuse=true

    # Allows to use stencils as blank patterns in the stencil table [default: false]
    B:allowStencilReuse=false

    # Allows you to use a piece of string as a binding [default: true]
    B:allowStringBinding=true

    # Creating a metal cast burns up the material that was used to create it [default: true]
    B:castingBurnsMaterial=true

    # Removes the ability to add modifiers with Gold, Diamond, Netherstars etc. [default: false]
    B:disableBonusModifierModifiers=false

    # Makes all non-TConstruct bows useless. You suddenly forgot how to use a bow. [default: false]
    B:disableRegularBows=true

    # Makes all non-TConstruct hoes to not be able to hoe ground. Use the Mattock. [default: false]
    B:disableRegularHoes=true

    # Makes all non-TConstruct swords useless. Like whacking enemies with a stick. [default: false]
    B:disableRegularSwords=true

    # Makes all non-TConstruct tools mine nothing [default: true]
    B:disableRegularTools=true

    # Stone Tools can only be used to create casts, but no tools [default: true]
    B:disableStoneTools=true

    # Allows to craft tool parts with a pattern and the material in any crafting grid. [default: false]
    B:easyPartCrafting=false

    # Allows to create Tool Station Tools (2-3 Parts) in any crafting grid [default: false]
    B:easyToolBuilding=false

    # Allows to also create Tool Forge Tools (4 Parts) in any crafting grid [default: false]
    B:easyToolBuildingForge=false

    # Allows to repair your tool in a crafting grid, without tool station [default: true]
    B:easyToolRepair=true

    # How many gravel are required to craft one Flint [range: 1 ~ 9, default: 3]
    I:gravelPerFlint=3

    # Silky Cloth needs gold ingots, instead of nuggets [default: true]
    B:moreExpensiveSilkyCloth=true

    # Silky Jewel needs an emerald block, instead of one emerald [default: false]
    B:moreExpensiveSilkyJewel=false

    # Flux modifier requires 2 Modifiers. Because that stuff is broken. [default: true]
    B:moreModifiersForFlux=true

    # Removes the random chance of getting flint from gravel [default: true]
    B:removeFlintDrop=true

    # Removes the creation of Obsidian in the Smeltery. This is normally used to prevent the early pre-diamond creation of obsidian tools. [default: true]
    B:removeObsidianAlloy=true

    # Removes the recipe for Tinker's Construct's stone torch [default: false]
    B:removeStoneTorchRecipe=false

    # Limits the amount how often a tool can be repaired. -1 means unlimited repairs, like normally. [range: -1 ~ 999, default: -1]
    I:repairsLimit=-1
}

(Modules.cfg)

# Configuration file

"tinker's construct addon: iguana tweaks for tinkers construct" {
    # Stuff used for debugging. You probably don't want this.
    B:Debug=false

    # Modify tool and item mining levels to create a tiered-ish progression
    B:HarvestLevelTweaks=false

    # All the Items Iguana Tweaks for TConstruct adds (Clay Buckets,...)
    B:Items=true

    # Adds additional MobHeads and control over MobHead drops.
    B:MobHeads=true

    # Makes Saw cut stuff again
    B:MultipartCompat=true

    # This module allows to override about any values relevant for TConstruct or IguanaTinkerTweaks.
    B:Override=false

    # Various Tweaks for vanilla Minecraft and Tinker's Construct. See Config.
    B:Restrictions=true

    # The Iguana Tweaks Leveling System for Tinker's Tools
    B:ToolLeveling=true

    # Replace parts of tools
    B:ToolPartReplacing=true

    # Various Tweaks for vanilla Minecraft and Tinker's Construct. See Config.
    B:Tweaks=true

    # This module takes care of additional awesome dungeon loot.
    B:WorldGen=true
}

Hope that helps.

**Comment by bonii-xx** [comment]
Yeah it's a bug with the vanilla diamond-modifier when the HarvestLevel Tweaks is deactivated. Known bug, will be fixed next version.

**Comment by jimmytrouser** [comment]
Thanks for the response.

Love the mod, can't play Minecraft without it.  Keep up the good work sir!

Thanks!

**Comment by Chiisato** [comment]
Has this issue been fixed in 2.1.2? "Fix incorrect stat detection and replacing with vanilla diamond/emerald modifier" according to the changelog.

I'm playing a pack with 2.1.2 and I added a emerald to a pickaxe and now I'm stuck with a incorrect data tool that won't update and I can't add any new modifiers to it (ie, using redstone uses the redstone but does not actually add haste/redstone to the tool since it's "stuck")

**Comment by Cuppycake** [comment]
Having this issue tonight. All newest versions. Confirmed it was the diamond and emerald modifiers. Any known fix?

**Comment by Zyin055** [comment]
I'm experiencing this bug in 2.1.2. It occurred after I changed the config file so that the level up requirement to mine higher ores was disabled. After the change I crafted a full bronze hammer and while it was still in the tool forge it said it could mine tin, but when I assembled the hammer it said it could mine redstone and that it needed to be updated because it had incorrect data. It still works as intended, just has that silly red text.
